src/main/java/frc/robot/Alignable.java

```diff
@@ -5,21 +5,23 @@
 import java.util.function.BiFunction;
 
 public enum Alignable {
-    AMP(0,0,(x,y)-> new Rotation2d(Math.PI / 2)),
-    SPEAKER(0.46,5.67,(x2,y2)-> new Rotation2d(RobotContainer.shouldFlip() ? 0 : Math.PI).plus(new Rotation2d(Math.atan(y2/x2))));
+    AMP(0,0,0,(x,y)-> new Rotation2d(Math.PI / 2)),
+    SPEAKER(0.46,16.46,5.67,(x2,y2)-> new Rotation2d(RobotContainer.isRedAlliance() ? 0 : Math.PI).plus(new Rotation2d(Math.atan(y2/x2))));
 
     private final double blueX;
+    private final double redX;
```

Comment:
What is redX and why do we need two x coords?

Reply:
I decided to make a separate red and blue x, instead of applying an offset like I did in FieldLocation because I thought there might be a situation where you want to a align with something that is not mirrored on the field. Upon further thought I cant think of such as situation. LMK if I should apply an offset instead

Reply:
That is fine. Once I understand what it is for, it makes sense. We may need redY once we have the full list of items to align on, but this is good for now.